tests/test_type_errors.py

```diff
@@ -0,0 +1,188 @@
+"""
+This file contains a test allowing assertions to be made that an expected type error is in
+fact produced by the type-checker. I.e. that the type checker is not delivering a false
+negative.
+
+To use the test, add a comment of the following form to your test code:
+
+    # assert-type-error-pyright: 'No overloads for "execute_operation" match' await
+    nexus_client.execute_operation(  # type: ignore
+
+The `type: ignore` is only necessary if your test code is being type-checked.
+"""
+
+import itertools
+import json
+import os
+import platform
+import re
+import subprocess
+import tempfile
+from pathlib import Path
+
+import pytest
+
+
+def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
```

Comment:
I bit confused here. How is this used? It seems to look for certain function names but ignores them if they are in this file but I can find them nowhere else? Is this code not leveraged in this PR?

Reply:
It's a pytest hook https://docs.pytest.org/en/stable/how-to/parametrize.html#pytest-generate-tests. Here we're using it to dynamically generate a test for every test file that contains type error assertion comments.

Reply:
It generates a test like this (we only have those comments in one file at the moment):

```
tests/test_type_errors.py::test_type_errors_pyright[test_type_checking_sync_operation.py] PASSED 
```